src/tests/multihost/sssd/testlib/common/utils.py

```diff
@@ -2020,36 +2020,52 @@ def __init__(self, ad_host):
 
     def get_zones(self):
         """ Returns a list of all the forward and reverse zones  on the server excluding msdcs """
-        zones = self.ad_host.run_command('dnscmd.exe /EnumZones | grep Primary | grep -v msdcs | cut -f2 -d" "')
-        return zones.stdout_text
+        zones = str.split(self.ad_host.run_command("dnscmd.exe /EnumZones").stdout_text, "\n")
+        i = 0
+        while 7 >= i:
+            del zones[0]
+            i += 1
+        i = 0
+        while 4 >= i:
+            del zones[len(zones)-1]
+            i += 1
 
     def print_zone(self, zone):
         """ Prints all the contents of a zone file, takes domain.com or 1.168.192.in-addr.arpa string """
-        zone = self.ad_host.run_command(f'dnscmd.exe /zoneprint {zone} | grep -v \; | sed "s/\t/ /g" | grep e |'
-                                        f'sed "s/\   / /g"')
-        return zone.stdout_text
+        zone_out = self.ad_host.run_command(f"dnscmd.exe /zoneprint {zone}")
+        return zone_out.stdout_text
 
     def find_a(self, hostname):
         """ Searches the zone for a specific A record, returning True for success and False for failure """
         domain = self.ad_host.realm.lower()
         host = hostname.split(".")[0]
-        results = self.ad_host.run_command(f'dnscmd.exe /zoneprint {domain} | grep -v \; | sed "s/\t/ /g" | grep e |'
-                                           f'sed "s/\   / /g" | grep {host}', raiseonerr=False)
-        if host in results.stdout_text:
-            return True
-        return False
+        seconds = 0
+        while seconds <= 300:
+            results = self.ad_host.run_command(f"dnscmd.exe /zoneprint {domain}")
+            if host not in results.stdout_text:
+                time.sleep(15)
+                seconds += 15
+            else:
+                return True
+
+            if seconds > 600 and host not in results.stdout_text:
```

Comment:
I'm not sure this will ever be true.  If host not in stdout, you add 15 until it reaches 300 then the while loop stops.
Do you need to change 300 to 600?

Also I think the while loop can be simplified some to eliminate the code scanning notice:

```
while seconds <= 300:
    results = self.ad_host.run_command(f"dnscmd.exe /zoneprint {domain}")
    if host in results.stdout_text:
        return True

    time.sleep(15)
    seconds += 15

return False
```